pinot-common/src/main/java/com/linkedin/pinot/common/data/FieldSpec.java

```diff
@@ -71,6 +73,9 @@
   @ConfigKey("defaultNullValue")
   private transient String _stringDefaultNullValue;
 
+  @ConfigKey("stringLengthLimit")
+  private int _stringLengthLimit = DEFAULT_STRING_LENGTH_LIMIT;
```

Comment:
Why call it string length limit? Can we not call it maxLength and use it for bytes as well. Also it’s better to check string. GetBytes size rt

Reply:
I don't know the behavior of trimming string in the middle of a UTF-8 encoded character. Also, we only support fixed length bytes for dictionary, so I don't know if we should trim bytes as well (then we might run into de-serialization exception issue).

Reply:
Since this is on a per-field basis, let us call it maxLength. We can choose not to implement it for BYTE_ARRAY since that can break encoding. And yes, agree on limiting the number of characters in string rather than number of encoded bytes.

Reply:
@fx19880617 disagree re maxBytes.  If we call it maxBytes it will somehow imply that we trim to that many bytes even if is a String. As pointed out by @Jackie-Jiang it does not make sense to trim bytes such that a String cannot be decoded. Let us call it maxLength and the semantics can be based on the data type. For ByteArray, it will translate to max length of the array.